CONTRIBUTING.md

```diff
@@ -1,3 +1,169 @@
 # Guidelines for contributing
 
-Coming soon...
+## General Workflow
+
+We are using [Git][git], [GitHub][github] and [Git Flow][git-flow].
+
+> **Note:** If you are a **beginner** and do not have a lot of experience with
+> this sort of workflow, please do not feel overwhelmed. We will guide you
+> through the process until you feel comfortable using it. And do not worry
+> about mistakes either - everybody does them. Often! Our project layout makes
+> it very very hard for anyone to cause irreversible harm, so relax, try things
+> out, take your time and enjoy the work! :)
+
+We would kindly ask you to abide by our [Code of Conduct][coc] in all
+interactions with the community when contributing to this project, regardless
+of the type of contribution. We will not accept any offensive or demeaning
+behavior towards others and will take any necessary steps to ensure that
+everyone is treated with respect and dignity.
+
+## Issue Tracker
+
+Please, use each project's GitHub [issue tracker][issue-tracker] to:
+
+- find issues to work on
+- report bugs
+- propose features
+- discuss future directions
+
+## Submitting Issues
+
+Please, choose a template when submitting an issue: choose the
+[**bug report** template][bug-report] only when reporting bugs; for all other
+issues, choose the [**feature request** template][feat-request]. Please follow
+the instructions in the templates.
+
+You do not need to worry about adding labels or milestones for an issue, the
+project maintainers will do that for you. However, it is important that all
+issues are written concisely, yet with enough detail and with proper
+references (links, screenshots, etc) to allow other contributors to start
+working on them. For bug reports, it is essential that they include all
+information required to reproduce the bug.
+
+Please **do not** use the issue tracker to ask usage questions, installation
+problems etc., unless they appear to be bugs. For these issues, please use
+the [communication channels](#communication) outlined below.
+
+## Communication
+
+Send us an [email][contact] if you want to reach out to us.
+
```

Comment:
Add a top-level (`##`) header here called something like "Contributing Code Changes" or something to that effect. Then put all the other `##` headers below, except the Co-Maintainer section.